Fix margin/padding styles for the `<AppButton>` when having spinner/icon/text

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

It's a follow-up fix from #918.

- Reset to the original `padding` when a `<AppButton>` contains both text and icon. 
    - https://github.com/woocommerce/google-listings-and-ads/pull/918#pullrequestreview-720450325
- Change the appending logic of loading spinner to prevent `<Button>` adding the unsuitable style to SVG icon. 
    - https://github.com/woocommerce/google-listings-and-ads/pull/918#discussion_r681924240

### Screenshots:

Revised button styles are shown on the left side. The right side is the current **trunk** version.

![Kapture 2021-08-17 at 18 00 33](https://user-images.githubusercontent.com/17420811/129707258-aec94f34-44a7-4d2e-beb0-cbb1d33b7fe1.gif)

💡  Please note that I added some examples in the demo GIF to show the difference. But currently, we don't have a use case that has text, icon, and loading state at the same time, such as the loading "Edit in Settings" button shown in the GIF. 

![Kapture 2021-08-17 at 17 40 04](https://user-images.githubusercontent.com/17420811/129711528-e14c611f-1a9e-4317-877a-ee9388471324.gif)

### Detailed test instructions:

1. Browse around to see if buttons are displayed properly.
2. Go to the edit contact info page in GLA Settings to see if:
    1. the refresh icon should be at the center of the clickable area
    ![image](https://user-images.githubusercontent.com/17420811/129710154-5281d8a3-88d8-4889-9aef-76cb65271975.png)
    2. there is a gap between the "Edit in Settings" text and the External Icon, and no `margin-right: 8px` on the icon.
    3. the padding of the "Edit in Settings" button should look nicer than before.
    ![image](https://user-images.githubusercontent.com/17420811/129709887-1e5a7875-2e28-41ef-a214-b56c584a2c04.png)

### Changelog entry

> Fix - Fix margin/padding styles for the AppButton when having spinner/icon/text

eason9487 committed:
Fix margin/paddding styles for AppButton when having loading/icon/text

ecgan commented on js/src/components/app-button/index.scss:
```diff
@@ -10,12 +10,19 @@
-	 * @see https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/components/src/button/style.scss#L309-L311
+	 * Reset the padding when a button contains both text and icon.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/components/src/button/style.scss#L292
+	 * @see https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/components/src/button/style.scss#L15
```
Thank you for the reference! 🙂  👍

ecgan commented on js/src/components/app-button/index.scss:
```diff
@@ -10,12 +10,19 @@
-	&--fix-icon-position-right.has-icon.has-text svg {
-		margin-right: 0;
-		margin-left: 8px;
+	&--fix-padding.has-icon {
```
I was thinking of just using `.app-button.has-icon` without the need of "--fix-padding", and I tested it out, it works for our case, but if we were to have a icon-only button without text, then the button would be in a slight rectangular shape instead of a square shape. 

💅  Instead of calling it "--fix-padding", how about "--icon-with-text"?

eason9487 replied:
I think it's ok to make its name more general instead of specifying the `padding` in it. Renamed by https://github.com/woocommerce/google-listings-and-ads/pull/963/commits/d98db6e187043be67448402bb7aa62289eee2999.

ecgan reviewed (approved):
LGTM, tested and working as expected 👍  . Left one minor 💅  comment.

eason9487 committed:
Rename the "app-button--fix-padding" to "app-button--icon-with-text"

Address https://github.com/woocommerce/google-listings-and-ads/pull/963#discussion_r692248592
